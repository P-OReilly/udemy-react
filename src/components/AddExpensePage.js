import React from 'react';
import ExpnseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { startAddExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startAddExpense(expense);
        this.props.history.push('/dashboard');
    };
    render() {
        return (
            <div>
                <h1>Add expense</h1>
                <ExpnseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);