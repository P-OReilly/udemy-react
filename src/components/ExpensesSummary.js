import { connect } from 'react-redux';
import React from "react";
import expensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount != 1 ? 'expenses' : 'expense';
    const formatedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (<div>
        <h1>Viewing {expenseCount} {expenseWord} totaling {formatedExpensesTotal}</h1>
    </div>
    )
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: expensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);