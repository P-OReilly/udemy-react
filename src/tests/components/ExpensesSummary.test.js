import React from 'react';
import { ExpensesSummary } from "../../components/ExpensesSummary";
import expenses from '../fixtures/expenses';
import { shallow } from 'enzyme';

test('Should render expense summary component correctly with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should render expense summary component correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={2355}/>);
    expect(wrapper).toMatchSnapshot();
});