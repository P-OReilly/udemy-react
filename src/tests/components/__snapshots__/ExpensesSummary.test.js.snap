// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense summary component correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totaling 
    $23.55
  </h1>
</div>
`;

exports[`Should render expense summary component correctly with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $2.35
  </h1>
</div>
`;
